src/test/java/org/broadinstitute/hellbender/engine/VariantWalkerIntegrationTest.java

```diff
@@ -107,4 +111,33 @@ public void testBestSequenceDictionary_FromVariantIndex() throws Exception {
         Assert.assertTrue(toolDict.getSequences().stream().allMatch(seqRec -> seqRec.getSequenceLength() == 0));
     }
 
+    @Test
+    public void testReadFilterOff() throws Exception {
+        final GATKTool tool = new TestGATKToolWithFeatures();
+        final File vcfFile = new File(publicTestDir + "org/broadinstitute/hellbender/engine/example_variants_WithReads.vcf");
+        final File bamFile = new File(publicTestDir + "org/broadinstitute/hellbender/engine/example_variants_WithReads.bam");
+        final String[] args = {
+                "--variant", vcfFile.getCanonicalPath(),
+                "--input", bamFile.getCanonicalPath(),
+                "--hasBackingReadSource", "true",
+                "--hasBackingReads", "true"
+        };
+        tool.instanceMain(args);
+    }
+
+    @Test
+    public void testReadFilterOn() throws Exception {
+        final GATKTool tool = new TestGATKToolWithFeatures();
+        final File vcfFile = new File(publicTestDir + "org/broadinstitute/hellbender/engine/example_variants_WithReads.vcf");
+        final File bamFile = new File(publicTestDir + "org/broadinstitute/hellbender/engine/example_variants_WithReads.bam");
+        final String[] args = {
+                "--variant", vcfFile.getCanonicalPath(),
+                "--input", bamFile.getCanonicalPath(),  // all reads have MAPQ == 0
+                "--readFilter", ReadFilterLibrary.MAPPING_QUALITY_NOT_ZERO.getClass().getSimpleName(),
+                "--hasBackingReadSource", "true",
+                "--hasBackingReads", "false"
+        };
+        tool.instanceMain(args);
+    }
```

Comment:
Would be better to have a test that only filters some of the reads, instead of just all or none. Also applies to `ReadsContextUnitTest` above. Eg., make an artificial read filter that only passes reads with certain names.